### PR TITLE
API: Allow rectifying Slave zones

### LINKED ISSUE #9066: Rectify via API not allowed for slave zones
- Program: Authoritative
 - Issue type: Bug report

### Short description

PowerDNS supports signing of slave zones. For this, slave zones need to be recitified. This works fine on incoming AXFR. If an existing slave zone is switched from unsigned to signed (signed on the PDNS slave), the zone needs to be rectified locally. This work fine with pdnsutil, but not via the REST API.

### Environment
 - Operating system: Ubuntu
 - Software version: 4.1 and trunk
 - Software source: self compiled

### Steps to reproduce
1. add a slave zone
2. add keys for the slave zone (we add them manually directly to cryptokeys table)
3. rectify via API

### Expected behaviour
Zone should be rectified

### Actual behaviour
Recitify is rejected: 
https://github.com/PowerDNS/pdns/blob/0ddde5f/pdns/ws-auth.cc#L1866

### Other information
A workaround would be to retransfer the zone, but this is unreliable as the master may not be available when DNSSEC is enabled for this zone.

### PR DESCRIPTION
**Pull request by zeha**
## Short description

Fixes #9066.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [x] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -1863,9 +1863,6 @@ static void apiServerZoneRectify(HttpRequest* req, HttpResponse* resp) {
   if (!dk.isSecuredZone(zonename))
```
Comment (mind04) [inline pdns/ws-auth.cc:1863]: Unrelated to the pull itself, but not allowing rectify for non dnssec zones is also unwanted.
I think we should remove this limitation as well.

Comment (zeha) [reply pdns/ws-auth.cc:1863]: Maybe this should actually be `if (!dk.isPresigned...`?

Comment (mind04) [reply pdns/ws-auth.cc:1863]: Yes, rectifying presigned zones will not end well.

Comment (zeha) [reply pdns/ws-auth.cc]: pushed followup commit for this

**Pull request merged**